### PR TITLE
Chi square method to calculate var and std dev CIs

### PR DESCRIPTION
**Pull request by JTPetter**
Implemented chi-squared method to calculate confidence intervals for std. dev. and variance in descriptives.

### CODE REVIEW

File: inst/qml/Descriptives.qml [hunk 1]
```diff
@@ -192,6 +192,20 @@ Form
...
 					label: qsTr("Width")
 				}
+
+				DropDown
+				{
+					name: "sdCiMethod"
+					label: qsTr("Method")
+					id: sdCiMethod
+					indexDefaultValue: 0
+					info: qsTr("How should the confidence interval be computed? By default, we use a chi-sqaured model. The alternative option is `Bootstrap`")
```
Comment (vandenman) [inline inst/qml/Descriptives.qml:202]:
```suggestion
					info: qsTr("How should the confidence interval be computed? By default, we use a chi-squared model. The alternative option is `Bootstrap`")
```
same below for variances.

I wonder if "Chi-squared model" is the right name. Perhaps we should just call this the "analytical" solution or "normal" approach and then leave the details for the help file?

File: R/descriptives.R [hunk 2]
```diff
@@ -1634,12 +1634,22 @@ DescriptivesInternal <- function(jaspResults, dataset, options) {
...
+    lowerBound <- CIs[1]
+    upperBound <- CIs[2]
+  } else if (options[["sdCiMethod"]] == "chiSquaredModel") {
+    sigma <- sd(data)
+    n <- length(data)
+    alpha <- 1 - ciWidth
+    chiSqLower <- qchisq(alpha / 2, df = n - 1)
+    chiSqUpper <- qchisq(1 - alpha / 2, df = n - 1)
+    lowerBound <- sqrt((n - 1) * sigma^2 / chiSqUpper)
+    upperBound <- sqrt((n - 1) * sigma^2 / chiSqLower)
```
Comment (vandenman) [inline R/descriptives.R:1651]:
```suggestion
    CIs <- stats::quantile(sds, probs = percentiles)
    lowerBound <- CIs[1]
    upperBound <- CIs[2]
  } else if (options[["sdCiMethod"]] == "chiSquaredModel") {
    sigma <- stats::sd(data)
    n <- length(data)
    alpha <- 1 - ciWidth
    chiSqLower <- stats::qchisq(alpha / 2, df = n - 1)
    chiSqUpper <- stats::qchisq(1 - alpha / 2, df = n - 1)
    lowerBound <- sqrt((n - 1) * sigma^2 / chiSqUpper)
    upperBound <- sqrt((n - 1) * sigma^2 / chiSqLower)
```
a (Wikipedia) reference to the equations as a comment would also be nice.

File: R/descriptives.R [hunk 3]
```diff
@@ -1648,12 +1658,22 @@ DescriptivesInternal <- function(jaspResults, dataset, options) {
...
+    percentiles <- (1 + c(-ciWidth, ciWidth)) / 2
+    CIs <- quantile(variances, probs = percentiles)
+    lowerBound <- CIs[1]
+    upperBound <- CIs[2]
+  } else if (options[["varianceCiMethod"]] == "chiSquaredModel") {
+    sigma <- sd(data)
+    n <- length(data)
+    alpha <- 1 - ciWidth
+    chiSqLower <- qchisq(alpha / 2, df = n - 1)
+    chiSqUpper <- qchisq(1 - alpha / 2, df = n - 1)
```
Comment (vandenman) [inline R/descriptives.R:1673]:
```suggestion
    CIs <- stats::quantile(variances, probs = percentiles)
    lowerBound <- CIs[1]
    upperBound <- CIs[2]
  } else if (options[["varianceCiMethod"]] == "chiSquaredModel") {
    sigma <- stats::sd(data)
    n <- length(data)
    alpha <- 1 - ciWidth
    chiSqLower <- stats::qchisq(alpha / 2, df = n - 1)
    chiSqUpper <- stats::qchisq(1 - alpha / 2, df = n - 1)
```

**Review by vandenman — CHANGES_REQUESTED** [review]

**Comment by JTPetter** [comment]
@vandenman Thank you, I have addressed your feedback.

File: R/descriptives.R [hunk 4]
```diff
@@ -1632,14 +1632,25 @@ DescriptivesInternal <- function(jaspResults, dataset, options) {
   ))
 }
 
+# Formula for chi-square confidence intervals for std. dev. and variance found here: https://www.itl.nist.gov/div898/handbook/prc/section2/prc231.htm
```
Comment (vandenman) [inline R/descriptives.R:1635]: This link gives a 503 service unavailable for me. Perhaps https://en.wikipedia.org/wiki/Standard_deviation#Confidence_interval_of_a_sampled_standard_deviation and https://en.wikipedia.org/wiki/Variance#Distribution_of_the_sample_variance are a better reference?

**Review by vandenman — CHANGES_REQUESTED** [review]
Looks good. Could you also add a file called `NEWS.md` and list as a new feature? Something like

- new feature: analytic confidence intervals for standard deviation and variance (PR: #312).

See for example https://github.com/jasp-stats/jaspReliability/blob/master/NEWS.md

**Review by vandenman — APPROVED** [review]

**Pull request merged**